src/actions/guides/database/migrations.cr

```diff
@@ -0,0 +1,231 @@
+
+    ```crystal
+    def migrate
+      alter :users do
+        remove :old_field
+      end
+    end
+    ```
+
+    ## Add column
+
+    To add a new column, you'll use the `add` method inside of a `[create](#create-table)` block.
+
+    ```crystal
+    create :users do
+      add email : String
+      add birthdate : Time
+      add login_count : Int32
+    end
+    ```
+
+    ### Datatypes
+
+    There are a few different datatypes that you can use to declare the column type.
+
+    * `String` - Maps to postgres `text`.
+    * `Int32` - Maps tp postgres `int`.
+    * `Int64` - Maps to postgres `bigint`.
+    * `Time` - Maps to postgres `timestamptz`.
+    * `Bool` - Maps to postgres `boolean`.
+    * `Float` - Maps to postgres `decimal`. With options for precision and scale.
+    * `UUID` - Maps to postgres `uuid`.
+
+    ### Advanced Options
+
+    The `add` method takes several options for further customization when adding a field.
+
+    * `index` - `false` by default. When `true`, this will create an index on this column.
+    * `using` - `:btree` by default. The index method to use when an index is created.
+    * `unique` - `false` by default. When `true`, postgres will enforce a unique constraint on this field.
+    * `default` - The default value to use for this column.
+
+    ```crystal
+    create :users do
+      add email : String, index: true, unique: true
+      add login_count : Int32, default: 0
+    end
+    ```
+
+    ## Remove column
+
+    The `remove` method must go in the `[alter](#alter-table)` block.
+
+    ```crystal
+    alter :users do
+      remove :middle_name
+      remove :last_login_ip
+    end
+    ```
+
+    ## Add index
+
+    The easiest way is to add the `index: true` option on the `add` method. However, if you're adding indicies after the table is created, you can use the `create_index` method.
+
+    ```crystal
+    def migrate
+      create_index :users, [:status], unique: false
+    end
+    ```
+
+    ## Remove index
+
+    Use the `drop_index` method to remove the index.
+
+    ```crystal
+    def migrate
+      drop_index :users, [:status]
+    end
+    ```
+
+    ## Associations
+
+    If your tables have a one-to-many or a many-to-many relation, you can use these methods to create a foreign key constraint, or remove that constraint from your tables.
+
+    ### Add belongs_to
+
+    The `add_belongs_to` method will create that foreign key constraint.
+
+    ```crystal
+    def migrate
+      create :comments do
+        # This will create an author_id column with an index and a foreign_key
+        # set to the users table.
+        #
+        # When the associated author is deleted, their comments are also deleted
+        # because we set on_delete: :cascade
+        add_belongs_to author : User, on_delete: :cascade
+      end
+    end
+    ```
+
+    You must include the `on_delete` option which can be one of `:cascade`, `:restrict`, `:nullify`, or `:do_nothing`.
```

Comment:
I think we should explain these options. Some context here https://github.com/luckyframework/lucky/issues/773#issuecomment-491416449